axelor-account/src/main/java/com/axelor/apps/account/service/AnalyticDistributionTemplateServiceImpl.java

```diff
@@ -53,4 +57,46 @@ public boolean validateTemplatePercentages(
     }
     return true;
   }
+
+  @Override
+  public void checkAnalyticDistributionTemplateCompany(
+      AnalyticDistributionTemplate analyticDistributionTemplate) throws AxelorException {
+    if (analyticDistributionTemplate.getCompany() != null) {
+      List<AnalyticDistributionLine> analyticDistributionLineList =
+          analyticDistributionTemplate.getAnalyticDistributionLineList();
+      boolean checkAxis = false;
+      boolean checkJournal = false;
+      for (AnalyticDistributionLine analyticDistributionLine : analyticDistributionLineList) {
+        if (analyticDistributionTemplate.getCompany()
+                != analyticDistributionLine.getAnalyticAxis().getCompany()
```

Comment:
analyticAxis is required only in view, it's safer to check the case where it is null here.